plugins/Files/js/sagas/helpers.js

```diff
@@ -219,9 +219,35 @@ export const parseUploads = (files) =>
 
 // Search `files` for `text`, excluding directories not in `path`
 export const searchFiles = (files, text, path) => {
-	let matchingFiles = List(files).filter((file) => file.siapath.indexOf(path) !== -1)
-	matchingFiles = matchingFiles.filter((file) => file.siapath.toLowerCase().indexOf(text.toLowerCase()) !== -1)
-	return matchingFiles
+	const filteredFiles = List(files)
+	  .filter((file) => file.siapath.indexOf(path) === 0 && file.siapath !== path)
+	  .filter((file) => file.siapath.toLowerCase().includes(text.toLowerCase()))
+
+	let parsedFiles = Map()
+	filteredFiles.forEach((file) => {
+		let type = 'file'
+		let name = Path.posix.basename(file.siapath)
+		let siapath = file.siapath
+		const pathComponents = file.siapath.split('/')
+		if (!pathComponents[pathComponents.length - 1].toLowerCase().includes(text.toLowerCase()) || file.siaUIFolder) {
```

Comment:
isn't `pathComponents[pathComponents.length - 1]` the same as `Path.posix.basename(file.siapath)`?